### PR TITLE
fix: Select refactoring known issues

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
The goal of this PR is to solve the known issues of the new Antd-based Select component, as follows:

- When showSearch is ON and labels are customized with JSX, the Select can only search by value. For instance, Explore - Customise - COLOR SCHEME - Search for airbnb - won’t work - Will only work with searching bnb as the value used is bnbColors
- When using labels with custom JSX the rendered HTML is showing label=[Object, Object]
- Write a string, then delete it entirely. The string is still there and selected
- With allowNewOptions true, type a non-existent option, then delete it char by char. Not the entire string is deleted.
- When trying to create a new option, but the search is returning matching options, you won’t be able to create the new option, as the Select will instead pick the first matching option

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Select/Select.tsx [hunk 1]
```diff
@@ -458,18 +415,26 @@ const Select = ({
...
+        optionFilterProps.every(prop => {
+          const optionProp = option?.[prop]
+            ? String(option[prop]).trim().toLowerCase()
+            : '';
+          if (optionProp.includes(searchValue)) {
+            found = true;
+            return false;
+          }
+          return true;
+        });
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Select/Select.tsx:433]:
```suggestion
        return optionFilterProps.some(prop => {
          const optionProp = option?.[prop]
            ? String(option[prop]).trim().toLowerCase()
            : '';
          return optionProp.includes(searchValue);
        });
```

File: superset-frontend/src/components/Select/Select.tsx [hunk 2]
```diff
@@ -549,6 +480,75 @@ const Select = ({
...
+    setSelectOptions(
+      options && Array.isArray(options) ? options : EMPTY_OPTIONS,
+    );
+  }, [options]);
+
+  useEffect(() => {
+    setSelectValue(value);
+  }, [value]);
+
+  useEffect(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Select/Select.tsx:494]: Should we only add a new option if `allowNewOptions` is true? Currently is adding even if `allowNewOptions` is false. If we do this, what will happen when `allowNewOptions` is false and an unknown value is passed? Should we discard it and override the `selectedValue`?

Comment (geido) [reply superset-frontend/src/components/Select/Select.tsx:494]: I believe it makes sense to keep it when `allowNewOptions` is both `false` and `true`. I don't see the danger in doing that but I see the danger in discarding a value. What do you think?

Comment (michael-s-molina) [reply superset-frontend/src/components/Select/Select.tsx:494]: Agreed

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the fixes!

**Pull request merged**